### PR TITLE
Remove bogus `null` arguments from emitters. Fixes gh-561

### PR DESCRIPTION
**Pull request by rwaldron**
Todo after landing: 

- Regenerate example programs
- Update wiki docs
  - Accelerometer
  - Gyro
  - Motor
  - Ping
  - Pir
  - Sensor
  - Sonar
  - Temperature

### CODE REVIEW

File: lib/ir.js [hunk 1]
```diff
@@ -78,11 +77,11 @@ Devices = {
       };
 
       // if ( temp.left < 200 ) {
-      //   this.emit( "left", err, timestamp );
```
Comment (BrianGenisio) [inline lib/ir.js]: Remove commented code?

File: lib/button.js [hunk 2]
```diff
@@ -126,7 +126,7 @@ function Button(opts) {
 
       timeout = setTimeout(function() {
         if (this.isDown) {
-          this.emit("hold", err);
+          this.emit("hold");
```
Comment (BrianGenisio) [inline lib/button.js]: Other examples of data-less events use date stamps.  Should we standardize on this?

**Comment by BrianGenisio** [comment]
LGTM.

**Comment by dtex** [comment]
Agreed, looks good to me to.

**Comment by divanvisagie** [comment]
I restarted the Travis build , lets see.

File: lib/ir.js (hunk 1 above)
Comment (rwaldron) [inline lib/ir.js]: Will do.

Comment (rwaldron) [inline lib/ir.js]: Actually, I think this whole file will be refactored soon, so I'm going to leave this in for now (out of sheer laziness)

File: lib/button.js (hunk 2 above)
Comment (rwaldron) [inline lib/button.js]: Unfortunately, it's a mixed collection of `new Date()` and `Date.now()` which will also have to be fixed, but in another patch

**Comment by rwaldron** [comment]
Rebased

**Pull request closed without merging**